training/src/app/modules/chuck-norris/components/search/search.component.ts

```diff
@@ -0,0 +1,78 @@
+import { Component, Output, EventEmitter, OnInit } from '@angular/core';
+import { AbstractControl, FormBuilder, FormGroup, Validators } from "@angular/forms";
+import { debounceTime, Subject, takeUntil, tap } from "rxjs";
+
+
+@Component({
+	selector: 'app-search',
+	templateUrl: './search.component.html',
+	styleUrls: ['./search.component.scss'],
+
+})
+export class SearchComponent implements OnInit {
+
+	public searchForm: FormGroup;
+	private componentDestroyed$: Subject<void>                        = new Subject();
+	@Output() private generateJokesOnClickMode: EventEmitter<string>  = new EventEmitter<string>();
+	@Output() private generateJokesOnSearchMode: EventEmitter<string> = new EventEmitter<string>();
+	@Output() private buttonShouldByDisable: EventEmitter<boolean>    = new EventEmitter<boolean>();
```

Comment:
the search component, doesnt "know" that there is a button somewhere else, so its not a good name. should be something like "isSearchValid"